sea-orm-macros/src/derives/model.rs

```diff
@@ -1,95 +1,74 @@
 use crate::{
     attributes::derive_attr,
-    util::{escape_rust_keyword, field_not_ignored, trim_starting_raw_identifier},
+    util::{escape_rust_keyword, trim_starting_raw_identifier},
 };
+use darling::{ast, util, FromDeriveInput, FromField};
 use heck::CamelCase;
 use proc_macro2::TokenStream;
-use quote::{format_ident, quote, quote_spanned};
-use std::iter::FromIterator;
-use syn::{punctuated::Punctuated, token::Comma, Ident, Lit, Meta};
+use quote::{format_ident, quote};
+use syn::Ident;
 
 enum Error {
-    InputNotStruct,
+    Parsing(darling::Error),
     Syn(syn::Error),
 }
 
+#[derive(Debug, FromField)]
+#[darling(attributes(sea_orm), allow_unknown_fields = true)]
+struct DeriveField {
+    ident: Option<Ident>,
+    #[darling(default)]
+    ignore: bool,
+    #[darling(default)]
+    enum_name: Option<String>,
+    #[darling(skip)]
+    column_ident: Option<Ident>,
+}
+
+#[derive(Debug, FromDeriveInput)]
+#[darling(supports(struct_named))]
 struct DeriveModel {
-    column_idents: Vec<syn::Ident>,
-    entity_ident: syn::Ident,
-    field_idents: Vec<syn::Ident>,
-    ident: syn::Ident,
-    ignore_attrs: Vec<bool>,
+    ident: Ident,
+    data: ast::Data<util::Ignored, DeriveField>,
+    #[darling(skip)]
+    entity_ident: Option<Ident>,
 }
 
 impl DeriveModel {
     fn new(input: syn::DeriveInput) -> Result<Self, Error> {
-        let fields = match input.data {
-            syn::Data::Struct(syn::DataStruct {
-                fields: syn::Fields::Named(syn::FieldsNamed { named, .. }),
-                ..
-            }) => named,
-            _ => return Err(Error::InputNotStruct),
+        let mut receiver = match DeriveModel::from_derive_input(&input) {
+            Ok(model) => model,
+            Err(error) => return Err(Error::Parsing(error)),
```

Comment:
Currently I cant match if this error is because of  deriving in `enum` because some error related enum is private in `darling` repo I have raised issue for that. https://github.com/TedDriggs/darling/issues/222#issue-1559510121

Reply:
Resolved with `daling` version `0.14.3` new error message is ```Unsupported shape `enum`. Expected struct with named fields.```